### PR TITLE
feat(medusa-file-s3): Allow all AWS options in config

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- Allow an aws config object to be passed as plugin config

### CODE REVIEW

**Review by adrien2p — APPROVED** [review]
Should we update the readme to reflect those changes?

File: docs/content/add-plugins/s3.md [hunk 1]
```diff
@@ -124,6 +124,27 @@ const plugins = [
...
+        bucket: process.env.S3_BUCKET,
+        aws_config_object: {
+          region: process.env.S3_REGION,
+          access_key_id: process.env.S3_ACCESS_KEY_ID,
+          secret_access_key: process.env.S3_SECRET_ACCESS_KEY,
+        }
+    },
+  },
+];
+```
```
Comment (shahednasser) [inline docs/content/add-plugins/s3.md:146]:
~~~suggestion
### Add AWS Configurations

You can pass AWS configurations in the plugin's options under the property `aws_config_object`. This property is an object that accepts [AWS Configurations](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html) as its properties.

For example, you can pass the `region`, `access_key_id` and `secret_access_key` in `aws_config_object`:

```jsx title=medusa-config.js
const plugins = [
  //...
  {
    resolve: `medusa-file-s3`,
    options: {
        s3_url: process.env.S3_URL,
        bucket: process.env.S3_BUCKET,
        aws_config_object: {
          region: process.env.S3_REGION,
          access_key_id: process.env.S3_ACCESS_KEY_ID,
          secret_access_key: process.env.S3_SECRET_ACCESS_KEY,
        }
    },
  },
];
```

Make sure to define `S3_REGION`, `S3_ACCESS_KEY_ID`, and `S3_SECRET_ACCESS_KEY` in your environment variables first.

~~~

**Review by shahednasser — CHANGES_REQUESTED** [review]
Made some changes to the documentation based on our writing/style guide, hope that's fine 😄. If there are inaccuracies feel free to make the necessary changes.

**Review by shahednasser — APPROVED** [review]

**Review by olivermrbl — APPROVED** [review]
LGTM!

**Pull request merged**